apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx

```diff
@@ -251,51 +280,14 @@ const SwapForm = () => {
       targetCurrency: targetCurrency?.name,
       partner: provider,
     });
+    if (isSwapLiveAppEnabled) {
+      setSwapWebProps(getSwapWebAppInputProps());
+      return;
+    }
 
     if (providerType === "DEX") {
-      const from = swapTransaction.swap.from;
-      const fromAccountId = from.parentAccount?.id || from.account?.id;
-
-      const pathname = `/platform/${getProviderName(provider).toLowerCase()}`;
-      const getAccountId = ({
-        accountId,
-        provider,
-      }: {
-        accountId: string | undefined;
-        provider: string;
-      }) => {
-        if (
-          !walletApiPartnerList?.enabled ||
-          !walletApiPartnerList?.params?.list.includes(provider)
-        ) {
-          return accountId;
-        }
-        const account = accounts.find(a => a.id === accountId);
-        if (!account) return accountId;
-        const parentAccount = isTokenAccount(account)
-          ? getParentAccount(account, accounts)
-          : undefined;
-        const walletApiId = accountToWalletAPIAccount(account, parentAccount)?.id;
-        return walletApiId || accountId;
-      };
-      const accountId = getAccountId({ accountId: fromAccountId, provider });
-      history.push({
-        // This looks like an issue, the proper signature is: push(path, [state]) - (function) Pushes a new entry onto the history stack
-        // It seems possible to also pass a LocationDescriptorObject but it does not expect extra properties
-        // @ts-expect-error so customDappUrl is not expected to be here
-        customDappUrl: providerURL,
-        pathname,
-        state: {
-          returnTo: "/swap",
-          accountId,
-        },
-      });
+      redirectToProviderApp(provider);
```

Comment:
I like this change. It will be easier to then do the switch with the FF to use our live-app instead